packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -52,6 +52,105 @@ async function offerToCreateTemplate(): Promise<void> {
   }
 }
 
+function findFoamVariables(templateText: string): string[] {
+  const regex = /\$(FOAM_[_a-zA-Z0-9]*)|\${(FOAM_[[_a-zA-Z0-9]*)}/g;
+  var matches = [];
+  const output: string[] = [];
+  while ((matches = regex.exec(templateText))) {
+    output.push(matches[1] || matches[2]);
+  }
+  const uniqVariables = [...new Set(output)];
+  return uniqVariables;
+}
+
+function getFoamTitle(givenValues: Map<string, string>) {
+  if (givenValues.has('FOAM_TITLE')) {
+    return Promise.resolve(givenValues.get('FOAM_TITLE'));
+  } else {
+    return window.showInputBox({
+      prompt: `Enter a title for the new note`,
+      value: 'Title of my New Note',
+      validateInput: value =>
+        value.trim().length === 0 ? 'Please enter a title' : undefined,
+    });
+  }
+}
+
+function slugify(input: string) {
+  // TODO: Replace this with the actual algorithm.
```

Comment:
you can use `github-slugger` (we already have the dependency)

Reply:
Wonderful! I've switched it to use that. 🎉